app/controllers/annotations_controller.rb

```diff
@@ -144,7 +152,11 @@ def update_annotation
     @content = params[:annotation_text][:content]
     @id = params[:annotation_text][:id]
     @submission_file_id = params[:submission_file_id]
```

Comment:
This is interesting. Your code change below adds a check for `params[:id]`, which is the annotation id. Is this new? Does this not always get passed? Certainly the ideal behaviour would have the annotation_id, rather than the text id, passed in. (For example, this would allow us to specify whether to change the text of a single annotation or all annotations with that text.)